docs/tokens/lockup.md

```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
 
-1. A regular account (also called "Owner Account" in context of lockups), let's say `user.near` or `3e52c197feb13fa457dddd102f6af299a5b63465e324784b22aaa7544a7d55fb`
-2. a lockup contract, with name `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`
+### Combinations
 
-Owner account is created first, either by following the NEAR Drop process or by creating new key pair using Trust, Ledger or another wallet.
+`v_start`, `v_cliff`, `v_end` are the aliases for vesting parameters; `l_start`, `l_end` are for lockup parameters.
+They could be easily transformed into initializing parameters described above.
 
-This owner account would usually have small balance 1-40 N to cover tx fees and the account storage itself (for those who are using 2FA in NEAR Wallet a larger amount is allocated, as it requires a multisig contract deployed on the account).
+![lockup_4](/docs/assets/lockup/lockup_4.png)
 
-Lockup contract is then deployed with predictable name. It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`. Which means that all lockup contracts end with `.lockup.near`.
+The resulting graph of the tokens becoming liquid is always the minimum between lockup and vesting mechanics.
 
-Lockup contract has few core properties:
+### Termination of vesting
 
-1. Total balance - amount of tokens at deployment when lockup started.
-2. Lockup start date - this is an absolute date when lockup can start unlocking. For example, for Coinlist participants this is `Oct 4, 2020 00:00 GMT`.
-3. Release length - after Phase 2 and lockup start date have passed, how long linear release will be lasting. For example, 1 year release length means that every day 1/365 unlocks.
- 
-If you have added or earned additional tokens above the initial balance, they are considered unlocked and can be freely withdrawn. For example any rewards that are earned using this lockup contract (eg from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.
+Vesting could be terminated by the foundation, an account configured at the moment of initializing the contract.
+It's important to understand how the termination works combining with the lockup schedule.
 
-The actual lockup release process happens on per block basis. E.g. if release length is 1 calendar year, it will actually be `31,536,000` seconds, and with ~1 second blocks, `~1/31,536,000` will be released per block.
+![lockup_5](/docs/assets/lockup/lockup_5.png)
 
-When the lockup has been fully released (lockup start date + release length has passed), the Owner Account can call to terminate the lockup and withdraw all the funds from it to the owner account.
+At the moment of termination, we fix the number of tokens that should be finally unlocked at the end.
+We continue to unlock the tokens as we normally do that by getting the minimum between lockup and vesting graph.
+We stop when we reach the desired amount of tokens.
```

Comment:
This sentence is unclear :thinking:

Reply:
Removed it, it does not give any value here